aten/src/ATen/native/mps/operations/UnaryKernel.mm

```diff
@@ -0,0 +1,147 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/native/DispatchStub.h>
+#include <ATen/native/UnaryOps.h>
+#include <ATen/native/mps/UnaryConstants.h>
+#include <torch/mps.h>
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Functions.h>
+#include <ATen/NativeFunctions.h>
+#else
+#include <ATen/ops/erfinv_native.h>
+#endif
+
+#include <fmt/format.h>
+
+namespace at::native {
+const std::string& getMetalType(const c10::ScalarType& t) {
+  // Mapping from c10::ScalarType to integral type that can be used for unary ops
+  static std::unordered_map<c10::ScalarType, std::string> scalar_to_metal_type = {
+      {c10::ScalarType::Half, "half"},
+      {c10::ScalarType::Float, "float"},
+      {c10::ScalarType::Long, "long"},
+      {c10::ScalarType::Int, "int"},
+      {c10::ScalarType::Short, "short"},
+      {c10::ScalarType::Bool, "bool"},
+      {c10::ScalarType::Char, "int8_t"},
+      {c10::ScalarType::Byte, "uint8_t"},
+  };
+
+  auto it = scalar_to_metal_type.find(t);
+  TORCH_CHECK(it != scalar_to_metal_type.end(), "Unsupported type ", t);
+  return it->second;
+}
+
+const std::string& getMetalType(const Tensor& t) {
+  return getMetalType(t.scalar_type());
+}
```

Comment:
they're not exactly identical as in bitwise operator it could be cast to "char" for 1 byte data type. In a numerical calculation like erfinv we need to cast 1 byte data type to uint8 or int8 or else the kernel will complain about doing math operations on "char".